### PR TITLE
Add basic doc to help contributers to add and test new modules

### PR DESCRIPTION
**Pull request by maleck13**
@rthallisey Basic doc outlining what I learned as I was writing the module. Not sure how much detail to go into about the broker etc.

### CODE REVIEW

File: docs/add-new-modules.md [hunk 1]
```diff
@@ -0,0 +1,39 @@
+# Adding New Modules
+
+This document will cover the basics of creating a new apb module and testing the module.
```
Comment (jmrodri) [inline docs/add-new-modules.md]: change apb to APB

File: docs/add-new-modules.md [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+
+## Writing the module
+
+Modules go under the library folder and should be named after the module they provide. For example the module for providing a last_operation update is named: ```asb_last_operation.py```
```
Comment (jmrodri) [inline docs/add-new-modules.md]: escape library with single backtick: `library`

**Review by rthallisey — APPROVED** [review]

**Comment by rthallisey** [comment]
thanks @maleck13

File: docs/add-new-modules.md [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+
+The module should be well documented with clear examples provided.
+
+Before writing a new module, it is recommended to create an issue on this repo outlining the functionality that you would like to see the module provide. This will promote discussion and a sharing of ideas to come up with the correct solution before any code has been written.
+
+## Testing the module
+
+The apb modules are intended to run from within the context of a Docker image as part of a POD running in a Kubernetes or Openshift environment. They regularly will need to interact with resources created within these environments. In order to test your module you will need to take the following steps.
```
Comment (jmrodri) [inline docs/add-new-modules.md]: apb -> APB
Openshift -> OpenShift

File: docs/add-new-modules.md [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+
+Example:
+
+``` 
+COPY asb_last_operation.py /etc/ansible/roles/ansibleplaybookbundle.asb-modules/library/ 
+```
+
+3) In your apb playbooks add calls to your new module.
+
+4) Build a new image and push it to your docker org
```
Comment (jmrodri) [inline docs/add-new-modules.md]: Are there commands you could include as examples to show how you built and pushed the image?

File: docs/add-new-modules.md [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+
+5) Ensure your ansible service broker is configured to look for apbs in the docker org where you have pushed the test apb
```
Comment (jmrodri) [inline docs/add-new-modules.md]: change ansible service broker to Ansible Broker.
apbs -> APBs

File: docs/add-new-modules.md [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+
+6) Perform the action that will call the playbook where your module is called from (provision,bind,update,deprovision, unbind)
+
+## Debugging 
+
+To debug the module, you can login to your cluster as a cluster admin then view the logs of the apb pod that is created, if there is a problem with your module, these logs should give you insight into what is happening.
```
Comment (jmrodri) [inline docs/add-new-modules.md]: I would include some sample commands. Sometimes I want to get into a technology without fulling knowing what's going on and need reminders of the commands to use.

```
oc login -u system:admin
oc project ansible-service-broker
oc logs asb-1-deadf
```
something like that would be helpful.

**Review by jmrodri — CHANGES_REQUESTED** [review]
Some terminology changes. Request for some additional example commands.

Excellent document. Super helpful.

**Comment by maleck13** [comment]
@jmrodri take 2 :)

**Comment by maleck13** [comment]
@jmrodri you ok with this doc now? I think it adds value and would be good to merge

**Comment by rthallisey** [comment]
It's looks like all the comments have been addressed and it's been around enough time for further comments.  I'm going to go ahead and merge it.

**Pull request merged**